shard.go

```diff
@@ -180,14 +179,31 @@ func marshalValues(values map[uint8]interface{}) []byte {
 		}
 
 		// Encode value after field id.
-		// TODO: Support non-float types.
 		switch v := v.(type) {
 		case float64:
+			buf = make([]byte, 9)
 			binary.BigEndian.PutUint64(buf[1:9], math.Float64bits(v))
+		case bool:
+			buf = make([]byte, 2)
```

Comment:
You shouldn't have to encode the type as that's kept in the metastore. Field id, followed by value (if a fixed size type) or in the case of bytes and strings, field id followed by length, followed by value.